src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/NewAzureKeyVaultManagedStorageAccountKey.cs

```diff
@@ -0,0 +1,87 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.KeyVault.Models;
+using System.Globalization;
+using System.Management.Automation;
+using KeyVaultProperties = Microsoft.Azure.Commands.KeyVault.Properties;
+
+namespace Microsoft.Azure.Commands.KeyVault
+{
+    [Cmdlet( VerbsCommon.New, CmdletNoun.AzureKeyVaultManagedStorageAccountKey,
+        SupportsShouldProcess = true,
+         ConfirmImpact = ConfirmImpact.High,
```

Comment:
This can be a destructive change as it regenerates the storage account key (like existing powershell cmdlet of New-AzureRmStorageAccountKey)

---
In reply to: [119998290](https://github.com/Azure/azure-powershell/pull/4061#discussion_r119998290) [](ancestors = 119998290)